tests/jest/actions.test.js

```diff
@@ -125,11 +146,39 @@ describe( 'getters', () => {
 			} );
 		} );
 
-		test.todo( 'Handles errors successfully' );
+		it( 'Handles fetch errors successfully', done => {
```

Comment:
I figured out why this test isn't working properly. There are two issues:

1. Your `expect` code is interrogating the `context.commit` mock in both cases. But `showCardStackMessage` is an action, (which gets called with `dispatch` instead of `commit`). So the first assertion should read `expect( context.dispatch ).toHaveBeenCalledWith( ... )`. The second assertion can remain unchanged, since `setPending` is a mutation.
2. It looks like you can only access the failure-based call in a `catch` block here, not in `always`. This must be some weird jQuery promise thing? I didn't expect it to work this way, but after some trial and error this code seemed to work properly. Note that a `then` handler is still required too:

```js
it( 'Handles fetch errors successfully', done => {
	var deferred = $.Deferred(),
		promise = deferred.promise();

	context.state.currentTab = 'popular';
	mockApi.get.mockReturnValue( promise );

	actions.getImages( context );

	promise.then( () => {
		// We only care about errors here, so do nothing
	} ).catch( () => {
		expect( context.dispatch ).toHaveBeenCalledWith( 'showCardStackMessage', {
			messageKey: 'machinevision-failure-message',
			type: 'error'
		} );
	} ).always( () => {
		expect( context.commit ).toHaveBeenCalledWith( 'setPending', {
			queue: 'popular',
			pending: false
		} );
		done();
	} );

	deferred.reject( {} );
} );
```